build: fix travis nightly build failure

tsachiherman opened the pull request:
## Summary

Travis nightly build failed due to no-output for over 10m from build process.

## Test Plan

Run on travis.

onetechnical reviewed (approved):
Looks fine, but do we think it'll pass in 30 minutes?

tsachiherman commented:
> Looks fine, but do we think it'll pass in 30 minutes?

No; I don't think that we'll pass the 30 minutes. I don't think that the tests would run for 90 minutes either.

Note that this is a high-watermark threshold. I already know that in some cases it clearly takes more than 10 minutes for the go-compiler build, so I figured that for the entire compilation ( including libsodium ), I should probably pad it generously.

winder reviewed (approved):
Find by me if we need it. Did something change in the build? I don't think I removed this, so the requirement is new.

tsachiherman commented:
> Find by me if we need it. Did something change in the build? I don't think I removed this, so the requirement is new.

It used to retry the entire execution 3 times.. and we moved that to the test phase only.
